ext/standard/string.c

```diff
@@ -1746,13 +1742,10 @@ PHP_FUNCTION(stristr)
 		Z_PARAM_BOOL(part)
 	ZEND_PARSE_PARAMETERS_END();
 
-	haystack_dup = estrndup(ZSTR_VAL(haystack), ZSTR_LEN(haystack));
-	orig_needle = estrndup(ZSTR_VAL(needle), ZSTR_LEN(needle));
-	found = php_stristr(haystack_dup, orig_needle, ZSTR_LEN(haystack), ZSTR_LEN(needle));
-	efree(orig_needle);
+	found = php_stristr(ZSTR_VAL(haystack), ZSTR_VAL(needle), ZSTR_LEN(haystack), ZSTR_LEN(needle));
```

Comment:
```suggestion
	found = php_memnistr(ZSTR_VAL(haystack), ZSTR_VAL(needle), ZSTR_LEN(haystack), ZSTR_LEN(needle));
```

Reply:
Note that the signature of `php_stristr` and `php_memnistr` isn't the same. One takes the length of the haystack and the other the a pointer to the end of the haystack. I kept it consistent with `zend_memnstr` but I do agree that one method would generally be enough if it does the same thing.

```c
const char *zend_memnistr(const char *haystack, const char *needle, size_t needle_len, const char *end);
char *php_stristr(char *s, char *t, size_t s_len, size_t t_len);
```

Reply:
Oh, right, never mind